homeassistant/components/vodafone_station/device_tracker.py

```diff
@@ -0,0 +1,133 @@
+"""Support for Vodafone Station routers."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from typing import Any
+
+from homeassistant.components.device_tracker import ScannerEntity, SourceType
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import _LOGGER, DEFAULT_DEVICE_NAME, DOMAIN
+from .coordinator import VodafoneStationDeviceInfo, VodafoneStationRouter
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up device tracker for Vodafone Station component."""
+
+    _LOGGER.debug("Start device trackers setup")
+    coordinator: VodafoneStationRouter = hass.data[DOMAIN][entry.entry_id]
+
+    tracked: set = set()
+
+    @callback
+    def update_router() -> None:
```

Comment:
Please name this `async_update_router` since it runs in the event loop and not the executor